tests/nnx/transforms_test.py

```diff
@@ -420,6 +420,58 @@ def f(m: nnx.Linear, x):
     y = compiled(m, x)
     self.assertEqual(m.count.value, 2)
 
+  def test_jit_static_no_static_in_shardings(self):
+    """Test nnx.jit with static args but no sharding for static positions."""
+    n_devices = jax.local_device_count()
+    devices = mesh_utils.create_device_mesh((n_devices,))
+    mesh = jax.sharding.Mesh(devices, ('data',))
+    
+    def fn(x, scale: float, use_relu: bool):
+      y = x * scale
+      if use_relu:
+        y = jnp.maximum(y, 0)
+      return y.sum()
+
+    # Use inputs with negative values so ReLU has an effect
+    x = jnp.linspace(-1.0, 1.0, 16, dtype=jnp.float32).reshape(4, 4)
+    scale = 0.5
+
+    x_sharding = jax.sharding.NamedSharding(mesh, jax.sharding.PartitionSpec('data'))
+    f = nnx.jit(fn, in_shardings=(x_sharding, None), static_argnums=(2,))
```

Comment:
We should test when `in_shardings` needs to be broadcast as well. Add another test case for this.